mods/gen4/random-teams.js

```diff
@@ -598,7 +596,7 @@ class RandomGen4Teams extends RandomGen5Teams {
 			item = 'Lustrous Orb';
 		} else if (ability === 'Guts') {
 			item = 'Toxic Orb';
-		} else if (slot === 0 && !counter['recoil'] && !counter['recovery'] && template.baseStats.hp + template.baseStats.def + template.baseStats.spd < 235) {
+		} else if (slot === 0 && !counter['recoil'] && !moves.some(id => !!recoveryMoves.includes(id)) && template.baseStats.hp + template.baseStats.def + template.baseStats.spd < 235) {
```

Comment:
The counter only counts the perfect "Recover" clones as recovery moves, excluding other recovery moves like Synthesis, Pain Split, and Rest. Those don't make sense with Sash either so I'm using my larger set of recovery moves. I'm not adding those moves to the counter's check since that affects all gens and the newer gens may have a good reason for not including them.